### PR TITLE
fix: insert on duplicate key update missing BindVars

### PR DESCRIPTION
**Pull request by wlx5575**
## Description
"In V18, the addition of trimming bound variables during insert needs to be included for each shard in this pull request: https://github.com/vitessio/vitess/pull/14064.  However, the scenario of 'insert on duplicate key' has been overlooked, and the bound variables added are not included for 'on duplicate key' variables."

## Related Issue(s)

- Fixes https://github.com/vitessio/vitess/issues/14729

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

### CODE REVIEW

**Comment by harshit-gangal** [comment]
@wlx5575 you need to run `make sizegen` and push it

**Comment by harshit-gangal** [comment]
Thank you @wlx5575 for your contribution and finding and fixing the bug ❤️

File: go/vt/vtgate/planbuilder/operator_transformers.go [hunk 1]
```diff
@@ -605,9 +605,7 @@ func generateInsertShardedQuery(ins *sqlparser.Insert) (prefix string, mids sqlp
 		ins.Table, ins.Columns)
 	prefix = prefixBuf.String()
 
-	suffixBuf := sqlparser.NewTrackedBuffer(dmlFormatter)
-	suffixBuf.Myprintf("%v", ins.OnDup)
-	suffix = suffixBuf.String()
+	suffix = ins.OnDup
```
Comment (harshit-gangal) [inline go/vt/vtgate/planbuilder/operator_transformers.go:608]: we need to walk the `onDup` and remove the database qualifier from it

Comment (wlx5575) [reply go/vt/vtgate/planbuilder/operator_transformers.go:608]: I use git cherry-pick 02bbbdd and it prompts me "bad revision '02bbbdd'". Where is the commit in the repository?😂

Comment (wlx5575) [reply go/vt/vtgate/planbuilder/operator_transformers.go]: I submitted a new version and made a judgment when removing database qualifier. Using your test case

**Review by harshit-gangal — APPROVED** [review]

File: go/test/endtoend/vtgate/misc_test.go [hunk 2]
```diff
@@ -27,6 +27,16 @@ import (
...
 )
 
+func TestInsertOnDuplicateKey(t *testing.T) {
+	conn, closer := start(t)
+	defer closer()
+
+	utils.Exec(t, conn, "insert into t11(id, sharding_key, col1, col2, col3) values(1, 2, 'a', 1, 2)")
+	utils.Exec(t, conn, "insert into t11(id, sharding_key, col1, col2, col3) values(1, 2, 'a', 1, 2) on duplicate key update id=10;")
+	utils.AssertMatches(t, conn, "select id, sharding_key from t11 where id=10", "[[INT64(10) INT64(2)]]")
+
```
Comment (frouioui) [inline go/test/endtoend/vtgate/misc_test.go:37]:
```suggestion
```

**Review by frouioui — APPROVED** [review]

**Pull request merged**